components/organisms/Repositories/repositories.tsx

```diff
@@ -55,7 +55,13 @@ const Repositories = ({ repositories }: RepositoriesProps): JSX.Element => {
         "/hub/insights/new"
       );
     } else {
-      signIn({ provider: "github" });
+      const reposIds = selectedRepos.map((repo) => repo.id);
+      signIn({ 
+        provider: "github",
+        options: {
+          redirectTo: `${window.location.origin}/hub/insights/new?selectedReposIDs=${JSON.stringify(reposIds)}&login=true&`,
```

Comment:
I don't think the `login` true is needed here. We can use the `access_token` to determine if its a returned auth request.